tensorzero-core/src/providers/gcp_vertex_gemini/optimization.rs

```diff
@@ -0,0 +1,692 @@
+            message: "Running".to_string(),
+            estimated_finish,
+            trained_tokens,
+            error: job.error,
+        },
+        GCPVertexGeminiFineTuningJobStatus::JobStateSucceeded => {
+            let tuned_model = job.tuned_model.as_ref().ok_or_else(|| {
+                Error::new(ErrorDetails::OptimizationResponse {
+                    message: "No tuned_model in Succeeded response".to_string(),
+                    provider_type: super::PROVIDER_TYPE.to_string(),
+                })
+            })?;
+
+            // Use endpoint if available, otherwise fall back to model ID
+            let model_name: String = tuned_model
+                .endpoint
+                .as_ref()
+                .unwrap_or(&tuned_model.model)
+                .clone();
+
+            let endpoint_id = tuned_model
+                .endpoint
+                .as_ref()
+                .and_then(|endpoint| endpoint.rsplit('/').next())
+                .map(|id| id.to_string());
+
+            let model_provider = UninitializedModelProvider {
+                config: UninitializedProviderConfig::GCPVertexGemini {
+                    model_id: None,
+                    endpoint_id,
+                    location,
+                    project_id,
+                    credential_location: Some(credential_location),
+                },
+                extra_headers: None,
+                extra_body: None,
+                timeouts: None,
+                discard_unknown_chunks: false,
+            };
+            OptimizationJobInfo::Completed {
+                output: OptimizerOutput::Model(UninitializedModelConfig {
+                    routing: vec![model_name.clone().into()],
+                    providers: HashMap::from([(model_name.clone().into(), model_provider)]),
+                    timeouts: TimeoutsConfig::default(),
+                }),
+            }
+        }
+        GCPVertexGeminiFineTuningJobStatus::JobStateFailed => OptimizationJobInfo::Failed {
+            message: "Failed".to_string(),
+            error: job.error,
+        },
+        GCPVertexGeminiFineTuningJobStatus::JobStateCancelling => OptimizationJobInfo::Failed {
+            message: "Cancelling".to_string(),
+            error: job.error,
+        },
+        GCPVertexGeminiFineTuningJobStatus::JobStateCancelled => OptimizationJobInfo::Failed {
+            message: "Cancelled".to_string(),
+            error: job.error,
+        },
+        GCPVertexGeminiFineTuningJobStatus::JobStatePaused => OptimizationJobInfo::Pending {
+            message: "Paused".to_string(),
+            estimated_finish,
+            trained_tokens,
+            error: job.error,
+        },
+        GCPVertexGeminiFineTuningJobStatus::JobStateExpired => OptimizationJobInfo::Failed {
+            message: "Expired".to_string(),
+            error: job.error,
+        },
+        GCPVertexGeminiFineTuningJobStatus::JobStateUpdating => OptimizationJobInfo::Pending {
```

Comment:
You can add an `GCPVertexGeminiFineTuningJobStatus` (see `fireworks_sft/mod.rs` for an example), and then use `job.state.to_string()` in a much smaller number of match arms